examples/refrigerator-app/silabs/src/RefrigeratorManager.cpp

```diff
@@ -0,0 +1,194 @@
+}
+
+void RefrigeratorManager::RefAndTempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value,
+                                                               uint16_t size)
+{
+    switch (attributeId)
+    {
+    case RefAndTempAttr::CurrentMode::Id: {
+        int16_t currentMode = static_cast<int16_t>(*value);
+        mCurrentMode        = currentMode;
+    }
+    break;
+
+    case RefAndTempAttr::StartUpMode::Id: {
+        int16_t startUpMode = static_cast<int16_t>(*value);
+        mStartUpMode        = startUpMode;
+    }
+    break;
+
+    case RefAndTempAttr::OnMode::Id: {
+        int16_t onMode = static_cast<int16_t>(*value);
+        mOnMode        = onMode;
+    }
+    break;
+
+    default: {
+        ChipLogError(AppServer, "Unhandled Refrigerator and Temperature attribute %ld", attributeId);
+        return;
+    }
+    break;
+    }
+}
+
+void RefrigeratorManager::TempCtrlAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value,
+                                                         uint16_t size)
+{
+    switch (attributeId)
+    {
+    case TempCtrlAttr::TemperatureSetpoint::Id: {
+        int16_t temperatureSetpoint = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mTemperatureSetpoint        = temperatureSetpoint;
+        TempCtrlAttr::TemperatureSetpoint::Set(endpointId, temperatureSetpoint);
+    }
+    break;
+
+    case TempCtrlAttr::MinTemperature::Id: {
+        int16_t minTemperature = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mMinTemperature        = minTemperature;
+        TempCtrlAttr::MinTemperature::Set(endpointId, minTemperature);
+    }
+    break;
+
+    case TempCtrlAttr::MaxTemperature::Id: {
+        int16_t maxTemperature = ConvertToPrintableTemp(static_cast<int16_t>(*value));
+        mMaxTemperature        = maxTemperature;
+        TempCtrlAttr::MaxTemperature::Set(endpointId, maxTemperature);
+    }
+    break;
+
+    default: {
+        ChipLogError(AppServer, "Unhandled Temperature controlled attribute %ld", attributeId);
+        return;
+    }
+    break;
+    }
+}
+
+void RefrigeratorManager::RefAlaramAttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value,
+                                                          uint16_t size)
+{
+    switch (attributeId)
+    {
+    case RefAlarmAttr::Mask::Id: {
+        auto mask = static_cast<uint32_t>(*value);
+        mState    = static_cast<chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap>(mask);
+        RefAlarmAttr::Mask::Set(endpointId, mMask);
+    }
+    break;
+
+    case RefAlarmAttr::State::Id: {
+        auto state = static_cast<uint32_t>(*value);
+        mState     = static_cast<chip::app::Clusters::RefrigeratorAlarm::AlarmBitmap>(state);
+        RefAlarmAttr::State::Set(endpointId, mState);
+    }
+    break;
+
+    case RefAlarmAttr::Supported::Id: {
```

Comment:
This is also a fixed attribute that should not change at runtime.

Reply:
removed this attribute update